### ISSUE TITLE
Unsupported argument; An argument named "expose" is not expected here.

### ISSUE THREAD

**Issue by caphrim007** [transcend-io/terragrunt-atlantis-config#167]
when running the config auto-generator, it seems that the go.mod terragrunt library is sufficiently out-of-date that it does not support the `expose` argument of the `include` block feature of terragrunt.

The following error is raised when terragrunt-atlantis-config encounters the attribute.

```
bash-5.1# terragrunt-atlantis-config generate --output atlantis.yaml --autoplan --parallel --create-workspace --create-project-name
INFO[0000] Could not find an old config file. Starting from scratch
Error: /atlantis/repos/cloudcoreo/vss-integration-service-infra/253/default/terraform/envs/dev-00/integration-service/terragrunt.hcl:3,3-9: Unsupported argument; An argument named "expose" is not expected here.
Usage:
  terragrunt-atlantis-config generate [flags]
```

This is triggered by the following terragrunt.hcl

```
include {
  path = find_in_parent_folders()
  expose = true
}
```

**Comment by caphrim007** [comment]
looking at when this functionality landed in Terragrunt, I think it was v0.30.4

**Comment by dmattia** [comment]
Perfect, this sort of thing is usually just solved by bumping up the go dep for terragrunt (as we use their parser directly).

I can look into this

**Comment by caphrim007** [comment]
@dmattia I'm a little out of my league on doing the actual implementation, but I can provide some more context in case it helps you.

There was also a recent release of 0.32.x

https://github.com/gruntwork-io/terragrunt/releases/tag/v0.32.0

Which included this release note

```
NOTE: This release is marked as backward incompatible, but there is no backward incompatible configuration changes. We expect all existing configuration to work as is. However, given the amount of changes internally to the include mechanism, we are marking this release as backward incompatible out of caution. If you encounter any issues using this release with existing configurations, please file an issue on this repo as the intention is to limit backward incompatibilities.

Added support for using multiple include blocks in a single configuration. Note that with this change, usage of a bare include block with no labels (e.g., include {}) is deprecated. It is recommended to update your configuration starting this release to attach a label to all include blocks (e.g., include "root" {}). You can learn more about multiple include blocks in the updated documentation.
```

This is more of just an FYI in case there are future issues raised about terragrunt-isms/features/fixes since it may require more work than just a version bump.

When I did an uninformed bump of the terragrunt dependency, I started getting this error when I ran the tests

```
$ go test ./...
# github.com/transcend-io/terragrunt-atlantis-config/cmd
cmd/parse_locals.go:80:19: invalid operation: includeConfig != nil (mismatched types config.TrackInclude and nil)
cmd/parse_locals.go:82:46: includeConfig.Path undefined (type config.TrackInclude has no field or method Path)
cmd/parse_locals.go:82:46: cannot use includeConfig (type config.TrackInclude) as type *config.IncludeConfig in argument to parseLocals
# github.com/transcend-io/terragrunt-atlantis-config/cmd [github.com/transcend-io/terragrunt-atlantis-config/cmd.test]
cmd/parse_locals.go:80:19: invalid operation: includeConfig != nil (mismatched types config.TrackInclude and nil)
cmd/parse_locals.go:82:46: includeConfig.Path undefined (type config.TrackInclude has no field or method Path)
cmd/parse_locals.go:82:46: cannot use includeConfig (type config.TrackInclude) as type *config.IncludeConfig in argument to parseLocals
```

Terragrunt seems to have added some new structures, or I'm simply not well-enough informed on how its supposed to work. Something to consider if you do the version bump.

https://github.com/gruntwork-io/terragrunt/blob/f02beeb7497faf7c030864b43fd56e5f5d879b04/config/config_helpers.go#L72

Lemme know if I can be of any more assistance testing things or providing more information. thanks.

**Comment by dmattia** [comment]
Thanks for that! Yeah, version bumps often rely on a few code changes, as this one does.

That's the downside of directly depending on Terragrunt's parsing logic, but then this library has the benefit of being able supporting exactly the same set of features as Terragrunt (up to the latest version I bump), which is worth the extra effort in my mind so that we can avoid bugs with differences in implementation over time.

**Comment by dmattia** [comment]
This terragrunt change does require some thought here ☹️ 

There used to only be one `include` allowed, but now there can be many. This means we need a merge strategy when two include blocks specify the same local value, which I'll need to think over.

**Comment by dmattia** [comment]
This should be fixed in v1.9.0: https://github.com/transcend-io/terragrunt-atlantis-config/releases/tag/v1.9.0

Please let me know if it's not 😄
